hack/lib/build/environment.sh

```diff
@@ -10,6 +10,28 @@ function os::build::environment::create() {
   set -o errexit
   local release_image="${OS_BUILD_ENV_IMAGE}"
   local additional_context="${OS_BUILD_ENV_DOCKER_ARGS:-}"
+  local tmpdir_template="openshift-env-XXXXXXXXXX"
+  if [[ -n "${TMPDIR:-}" ]]; then
+    # If TMPDIR is specified, respect it
+    local container_tmpdir=$(mktemp -d --tmpdir ${tmpdir_template})
```

Comment:
don't mix scoping statements (`local`) with expressions that evaluate, like subshells